reverse_iter_test.go

```diff
@@ -0,0 +1,148 @@
+package iradix
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestReverseIterator_SeekReverseLowerBound(t *testing.T) {
+	r := New()
+	keys := []string{"001", "002", "005", "010", "200", "201"}
```

Comment:
This is a good basic test but I suggest implementing a fuzz test to exercise all the edge cases more thoroughly.

I _guess_ that this case is not actually creating a tree with more than a single node depth which possibly explains why the apparent bug i noted isn't being triggered.

Of course I could just be wrong about that bug, but regardless I found _lots_ of bugs in the original LowerBound implementation from `TestIterateLowerBoundFuzz` that a sinple test like this would not have picked up.

Reply:
That's a nice test! I somehow missed it among the others, but I will add that and see if catches the bug you found.